frontend/task/libs/smart_contract/SmartContractView.tsx

```diff
@@ -35,12 +35,29 @@ export function SmartContractView() {
                 });
             });
         });
+        let balances = {};
+        processedLog.forEach((log) => {
+            if (log.amount) {
+                if ('origination' == log.kind) {
+                    balances[log.address] = Number(log.amount);
+                }
+                if (log.source && (undefined !== balances[log.source] || !isContract(log.source))) {
+                    // if undefined, it's not a contract and starts with 10000 tez
+                    balances[log.source] = (balances[log.source] ?? 10000) - Number(log.amount);
+                }
+                if (log.destination && (undefined !== balances[log.destination] || !isContract(log.destination))) {
+                    // if undefined, it's not a contract and starts with 10000 tez
+                    balances[log.destination] = (balances[log.destination] ?? 10000) + Number(log.amount);
+                }
+            }
+            log.balances = Object.assign({}, balances);
```

Comment:
```suggestion
            log.balances = {...balances};
```

You could use the ES6 spread syntax, shorter and easier to read ;)

Reply:
👍